genomepy/provider.py

```diff
@@ -804,7 +964,10 @@ class NCBIProvider(ProviderBase):
     assembly_url = "https://ftp.ncbi.nlm.nih.gov/genomes/ASSEMBLY_REPORTS/"
 
     def __init__(self):
-        self.genomes = None
+        # Necessary for bucketcache, otherwise methods will identical names
+        # from different classes will use the same cache :-O!
+        self.name = "NCBI"
+        self.genomes = self._get_genomes()
```

Comment:
This is a bit annoying as it starts the moment you begin using genomepy.
Could NCBIs _get_genomes() be run in the background when first called, and only be waited upon when using ncbi features?

I checked, and it could perhaps be done by
* threading
* asynchio

PS I was afraid this would happen for all providers, but it seems the others are really nice and quick!